### ISSUE TITLE
macOS versions always generates ganttproject.log in user folder

### ISSUE THREAD

**Issue by EGDBE** [bardsoftware/ganttproject#1717]
Everytime I open GanttProject in macOS Catalina it generates the file "ganttproject.log" in my user folder root. It would be great to prevent it. This is not the place for log files.

**Comment by dbarashev** [comment]
On the other hand, it is a place which surely exists, surely writable and relatively easy to find for an average macOS user. What's so terribly wrong with the log file in user's $HOME directory?

**Comment by juananpe** [comment]
Maybe another solution could be to add a .ganttproject _directory_ in $HOME and save there the two files that GanttProject generates right now (.ganttproject _file_ - an XML file with options and settings - and the .ganttproject.log file)

**Comment by dbarashev** [comment]
Ironically, @EGDBE doesn't mind `.ganttproject` file being in the user folder. Is it because home folder is the right place for config files or just because it is hidden?

**Comment by EGDBE** [comment]
1) If all apps did put its logs at home directory, it would become a jungle.

2) One reasonable candidate to put the log is in the own app directory: "/Applications/GanttProject.app" because:

- An average user should know that right clicking the app icon and picking "Show Package Contents" will navegate through app container. 

- Furthermore, this directory always will exists and the app will have the write/read perms guaranteed (own sandbox).

Thanks

**Comment by EGDBE** [comment]
@dbarashev no config or log file should be at home directory (regardless it's hidden or visible). As I said, the proper path is the **own app container**.

Thanks

**Comment by dbarashev** [comment]
> the proper path is the own app container.

May I kindly ask you to provide any proofs of that?

From my side, I can tell that the proper path is `~/Library/Logs`. Proofs: 

* https://www.howtogeek.com/356942/how-to-view-the-system-log-on-a-mac/
* https://stackoverflow.com/questions/49145655/mac-os-x-where-app-can-create-a-log-file

For preferences, ~/Library/Preferences is the right place: https://stackoverflow.com/questions/410013/where-do-osx-applications-typically-store-user-configuration-data

I admit it might be better than just plain user home. Although ~/Library is hidden in the Finder and thus requires additional instructions for locating, it seems it can be viewed with Console app which is ok.

**Comment by EGDBE** [comment]
I agree "~/Library/Logs" and "~/Library/Preferences" are good places.

Thanks
